toolchain/check/testdata/interface/fail_assoc_const_not_constant.carbon

```diff
@@ -0,0 +1,32 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+interface I {
+  // CHECK:STDERR: fail_assoc_const_not_constant.carbon:[[@LINE+3]]:7: ERROR: Pattern in associated constant declaration must be a single `:!` binding.
+  // CHECK:STDERR:   let a: i32;
+  // CHECK:STDERR:       ^
+  let a: i32;
+}
```

Comment:
As part of this, you poison the name scope. Would it be worth testing the result?

```suggestion
  // CHECK:STDERR: fail_assoc_const_not_constant.carbon:[[@LINE+3]]:7: ERROR: Pattern in associated constant declaration must be a single `:!` binding.
  // CHECK:STDERR:   let a: i32;
  // CHECK:STDERR:       ^
  let a: i32;
}

// Should fail silently.
alias Invalid = I.Nonexistent;
```